src/summary/documentation/dimensions-card.component.tsx

```diff
@@ -0,0 +1,82 @@
+import React from "react";
+import SummaryCard from "../cards/summary-card.component";
+import { match } from "react-router";
+import { getPatientDimensions } from "./dimensions.resource";
+import SummaryCardRow from "../cards/summary-card-row.component";
+import VerticalLabelValue from "../cards/vertical-label-value.component";
+import SummaryCardRowContent from "../cards/summary-card-row-content.component";
+import HorizontalSectionCard from "./horizontal-section-card.component";
+import DimensionsSectionCard from "./dimensions-card-row.component";
+import { ageConvert } from "./date-helper";
+import ShowMoreCard from "./show-more-card.component";
+import style from "./dimensions-card.css";
+export default function DimensionsCard(props: DimensionsCardProps) {
+  const [dimensions, setDimensions] = React.useState([]);
+  const [showElements, setShowElements] = React.useState(3);
+
+  React.useEffect(() => {
```

Comment:
This effect is rather complicated. I feel like we could simplify it substantially. Please do the following:

1. Paste an example response for the fhir observations endpoint so that I can see what the data looks like. I tried to find a patient with observations on openmrs-spa.org but wasn't able to.
2. Explain to me what you're trying to do with all the filters, maps, and pushes. Are you just trying to group the observations into Weight, Height, and BMI arrays?

Reply:
https://pastebin.com/PAzHTKkz
The fhir endpoint that return entries by code is not ready yet so I use the one that return all entries, I filter the ones that I want to use and then i group entries by date (are sorted by date as far as i know).

Reply:
Make this effect simpler. The code here is too complicated and unclear